src/core/nis1/Nis1KeyPair.ts

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2021 NEM
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import * as Crypto from 'crypto';
+import { Key } from '../Key';
+import { KeyPair } from '../KeyPair';
+import { keccakHash, KeccakHasher } from '../utils/Utilities';
+/* eslint @typescript-eslint/no-var-requires: "off" */
+const Ed25519 = require('./external/nacl-fast.js').lowlevel;
+
+export class Nis1KeyPair extends KeyPair {
+    /**
+     *Constructor
+     * @param {string} privateKey Private Key
+     */
+    constructor(privateKey: Key) {
+        super(privateKey);
+    }
+
+    /**
+     * Generate a random new keypair
+     * @returns {KeyPair} New keypair
+     */
+    public static generate(): Nis1KeyPair {
+        return new Nis1KeyPair(new Key(Crypto.randomBytes(32)));
+    }
+
+    /**
+     * @property Public key
+     * @returns {string} Raw public key string
+     */
+    public getPublicKey(): Key {
```

Comment:
i think this should be protected after @rg911 change. also, probably derivePublicKey is more correct.